### PR TITLE
test: add test for datalist autofills

### PR DESCRIPTION
**Pull request by loc**
#### Description of Change
I noticed that there weren't any tests for the autofill popup, which has a rather large (and now divergent) code footprint, so I added a simple one. 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: spec/fixtures/pages/datalist.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+</head>
+<body>
+  <input type="text" list="guitarists" />
+  <datalist id="guitarists">
+    <select>
+      <option value="John Mayer"></option>
+      <option value="Eric Clapton"></option>
+      <option value="Django Reinhardt"></option>
+    </select>
+   </datalist>
```
Comment (nornagon) [inline spec/fixtures/pages/datalist.html:13]:
```suggestion
  </datalist>
```

File: spec/fixtures/pages/datalist.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+<head>
+</head>
```
Comment (nornagon) [inline spec/fixtures/pages/datalist.html:4]:
```suggestion
```

File: spec-main/autofill-spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  })
+
+  afterEach(() => {
+    w.destroy()
+    w = null as unknown as BrowserWindow
+  })
+
+  it('can be selected via keyboard', async () => {
+    w.webContents.sendInputEvent({ type: 'keyDown', keyCode: 'Tab' })
+    await delay(100);
```
Comment (nornagon) [inline spec-main/autofill-spec.ts:29]: Is this delay really necessary?

File: spec-main/autofill-spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      }
+    })
+
+    await w.loadFile(path.join(fixturesPath, 'pages', 'datalist.html'))
+  })
+
+  afterEach(() => {
+    w.destroy()
+    w = null as unknown as BrowserWindow
+  })
```
Comment (nornagon) [inline spec-main/autofill-spec.ts:25]: I recommend the "new style" of test fixture, which is to create browserwindows in your test as needed and add `afterEach(closeAllWindows)` as cleanup.

Also, there's no particular need for `nodeIntegration` in this window I don't think.

File: spec-main/autofill-spec.ts (hunk 3 above)
Comment (loc) [reply spec-main/autofill-spec.ts]: Removed

File: spec-main/autofill-spec.ts (hunk 4 above)
Comment (loc) [reply spec-main/autofill-spec.ts]: 👍 done!

File: spec/fixtures/pages/datalist.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (nornagon) [inline spec/fixtures/pages/datalist.html:1]: Please put this file in `spec-main/fixtures`

**Pull request merged**